File: challenges/535_EncodeAndDecodeTinyURL/main.ts
/**
 * Encodes a URL to a shortened URL.
 */
function encode(longUrl: string): string {
  return btoa(longUrl);
}

/**
 * Decodes a shortened URL to its original URL.
 */
function decode(shortUrl: string): string {
  return atob(shortUrl);
}

console.log(decode(encode("https://leetcode.com/problems/design-tinyurl")));

File: challenges/535_EncodeAndDecodeTinyURL/main.cpp
#include <iostream>
#include <string>

class Solution
{
public:
    // Encodes a URL to a shortened URL.
    std::string encode(std::string longUrl)
    {
        return longUrl;
    }

    // Decodes a shortened URL to its original URL.
    std::string decode(std::string shortUrl)
    {
        return shortUrl;
    }
};

int main()
{
    Solution solution;

    std::cout << solution.decode(solution.encode("https://leetcode.com/problems/design-tinyurl"));

    return 0;
}
